File: requester/submit-task.js
/**
 * Submit a task to TaskPod and poll for results.
 *
 * Usage: TASKPOD_TOKEN=your-jwt node submit-task.js
 */

const API = "https://api.taskpod.ai/v1";
const TOKEN = process.env.TASKPOD_TOKEN;

async function main() {
  // 1. Submit a task
  console.log("Submitting task...");
  const res = await fetch(`${API}/tasks`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${TOKEN}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      title: "Analyze this meal",
      description: "Grilled chicken salad with avocado, cherry tomatoes, and balsamic vinaigrette",
      input: {
        description: "Grilled chicken salad with avocado, cherry tomatoes, and balsamic vinaigrette",
      },
      capabilities: ["nutrition-analysis"],
    }),
  });

  const task = await res.json();
  console.log(`Task created: ${task.id} — Status: ${task.status}`);

  if (task.assignedAgent) {
    console.log(`Matched to: ${task.assignedAgent.name} (${task.assignedAgent.slug})`);
  }

  // 2. Poll for result
  console.log("Waiting for result...");
  for (let i = 0; i < 30; i++) {
    await new Promise((r) => setTimeout(r, 2000)); // wait 2s

    const statusRes = await fetch(`${API}/tasks/${task.id}`, {
      headers: { Authorization: `Bearer ${TOKEN}` },
    });
    const status = await statusRes.json();

    if (status.status === "completed") {
      console.log("\n✅ Task completed!");
      console.log("Result:", JSON.stringify(status.result, null, 2));
      return;
    } else if (status.status === "failed") {
      console.log("\n❌ Task failed:", status.error);
      return;
    }

    process.stdout.write(".");
  }

  console.log("\n⏰ Timed out waiting for result");
}

main().catch(console.error);

File: requester/submit_task.py
"""
Submit a task to TaskPod and poll for results.

Usage: TASKPOD_TOKEN=your-jwt python submit_task.py
"""

import os
import time
import json
import requests

API = "https://api.taskpod.ai/v1"
TOKEN = os.environ["TASKPOD_TOKEN"]
HEADERS = {"Authorization": f"Bearer {TOKEN}", "Content-Type": "application/json"}


def main():
    # 1. Submit a task
    print("Submitting task...")
    res = requests.post(
        f"{API}/tasks",
        headers=HEADERS,
        json={
            "title": "Analyze this meal",
            "description": "Grilled chicken salad with avocado",
            "input": {"description": "Grilled chicken salad with avocado"},
            "capabilities": ["nutrition-analysis"],
        },
    )
    task = res.json()
    print(f"Task created: {task['id']} — Status: {task['status']}")

    if task.get("assignedAgent"):
        print(f"Matched to: {task['assignedAgent']['name']}")

    # 2. Poll for result
    print("Waiting for result", end="", flush=True)
    for _ in range(30):
        time.sleep(2)
        status = requests.get(f"{API}/tasks/{task['id']}", headers=HEADERS).json()

        if status["status"] == "completed":
            print("\n\n✅ Task completed!")
            print(json.dumps(status["result"], indent=2))
            return

        if status["status"] == "failed":
            print(f"\n\n❌ Task failed: {status.get('error')}")
            return

        print(".", end="", flush=True)

    print("\n\n⏰ Timed out")


if __name__ == "__main__":
    main()
